content/articles/building-and-integrating-spring-interceptor-in-java/index.md

```diff
@@ -0,0 +1,124 @@
+In my personal project, I have had reasons to add an `Interceptor`. As the name suggests, an `Interceptor` is a class that comes in between a request reaching the intended endpoint and a response returning to the requesting client - user or application. 
+
+With this, an interceptor can check a request before sending it to its destination. At this point, we might want to perform some authentication of headers, validations of the request body or simply modify the request object by adding some more fields or data to the request body.
+
+The same can be done for all responses coming from the endpoints. We can intercept them and modify them before passing them to the client. The interceptor is also a perfect place to build an encryption gateway. However, in this tutorial, we will only build a request/response interceptor while we will breeze through building an encryption/decryption gateway in the same interceptor.
+
+### Prerequisites
+- Familiarity with Java and Spring Boot
+- Familiarity with servlets and adapters
+- The IntelliJ IDE
+- An understanding of requests and responses
+
+### Table of contents
+- [Introduction](#introduction)
+- [Introducing the Methods for (Request/Response) Interception](#interceptor-methods)
+- [Registering a Custom Interceptor](#registering-a-custom-interceptor)
+- [Key Takeaways](#key-takeaways)
+- [Conclusion](#conclusion)
+
+### Introduction
+To get an understanding of how an `Interceptor` works, we need to know the workings of the `HandlerMapping` in Spring. What a `HandlerMapping` does is map a URL to a handler method. By doing this, a servlet called the `Dispatcher Servlet` can initiate it during request processing. The `Dispatcher Servlet` employs `HandlerAdapter` to run a call to the method. Meanwhile, an interceptor intercepts requests and performs some actions on them. They also ensure handler codes are not repeated across the project.
+
+For us to use Interceptors in Spring Boot, we need to add the web and Lombok dependencies in our `pom.xml` configuration file. The web dependency allows us to interact with services over HTTP while Lombok gives us access to the `@Slf4j` annotation which we will use later for logging.
+
+```xml
+<dependency>
+    <groupId>org.springframework</groupId>
+    <artifactId>spring-web</artifactId>
+    <version>5.3.13</version>
+</dependency>
```

Comment:
I'm going to be changing this to the `spring-boot-starter-web` dependency. This dependency isn't enough to get all the libraries you are using.

Reply:
Oh indeed you are right. sensational miss. Thank you